<?xml version="1.0"?>

<launch>
  <arg name="geotiff_map_file_path" default="$(find hector_geotiff)/maps"/>
  <param name="/use_sim_time" value="false"/>

  <!-- <node pkg="tf" type="static_transform_publisher" name="static_tf_world_map"
        args="1 0 0 0 0 0 /world /map 100"/> -->

  <node pkg="rviz" type="rviz" name="rviz"
        args="-d $(find hector_slam_launch)/rviz_cfg/mapping_demo.rviz"/>

  <!--<include file="$(find ros_x_habitat)/launch/hector_mapping_default.launch"/>-->

  <!-- <node pkg="tf" type="static_transform_publisher" name="base_to_laser"
        args="0 0 0 0 0 0 /base_frame /laser 100"/> -->

  <node pkg="tf" type="static_transform_publisher" name="map_to_odom"
        args="0 0 0 0 0 0 /map /odom 100"/>
  <!-- <node pkg="tf" type="static_transform_publisher" name="base_frame_to_laser" args="0.0 0.0 0.0 -1.5708 0.0 -1.5708 /base_frame /laser 50" /> -->

  <node pkg="nodelet" type="nodelet" name="rgbd_sync"
        args="standalone rtabmap_sync/rgbd_sync" output="screen">
    <remap from="rgb/image"       to="rgb"/>
    <remap from="depth/image"     to="depth"/>
    <remap from="rgb/camera_info" to="camera_info"/>
    <remap from="rgbd_image"      to="rgbd_image"/>
    <param name="approx_sync" value="true"/>
  </node>

  <node name="rtabmap" pkg="rtabmap_slam" type="rtabmap"
        output="screen" args="--delete_db_on_start">
    <param name="frame_id"        value="base_frame"/> <!--base_frame-->
    <param name="odom_frame_id"   value="odom"/> <!--odom-->
    <param name="map_frame_id"    value="map"/>

    <param name="publish_tf"          value="false"/>
    <!-- <param name="publish_tf"          value="true"/> -->
    <param name="publish_odometry_tf" value="false"/> <!--false-->

    <param name="subscribe_depth" value="false"/>
    <param name="subscribe_rgbd"  value="false"/> <!--true-->
    <param name="subscribe_rgb" value="false"/>
    <param name="subscribe_scan"  value="true"/> <!--true-->
    <param name="subscribe_odom" value="true"/>
    <param name="queue_size"      value="10"/>

    <remap from="odom"       to="odom"/>
    <remap from="scan"       to="scan"/>
    <remap from="rgbd_image" to="rgbd_image"/>

    <param name="RGBD/ProximityBySpace" value="false"/>
    <param name="Kp/LoopThr" value="0.999"/> 

    <param name="Optimizer/Strategy" value="0"/>

    <param name="Reg/Strategy"              value="0" /> 
    <param name="Reg/Force3DoF"             value="true"/>
    <param name="RGBD/AngularUpdate"        value="0.05"/>
    <param name="RGBD/LinearUpdate"         value="0.05"/>
    <param name="RGBD/CreateOccupancyGrid"  value="true"/>
    <param name="Grid/3D"                   value="false"/>
    <!-- <param name="Grid/MaxObstacleRange" value="3"/> -->
    <param name="Grid/RangeMax" value="3.5"/>
    <param name="Grid/RangeMin" value="0.05"/>
    <param name="Grid/ClosingRadius" value="0.3"/>

    <!-- <param name="RGBD/NeighborLinkRefining" value="true"/>
    <param name="RGBD/ProximityBySpace"     value="true"/>
    <param name="RGBD/AngularUpdate"        value="0.05"/>
    <param name="RGBD/LinearUpdate"         value="0.05"/>
    <param name="RGBD/OptimizeFromGraphEnd" value="false"/>
    <param name="Grid/FromDepth"            value="false"/>
    <param name="Reg/Force3DoF"             value="true"/>
    <param name="map_always_update"  value="true"/>
    <param name="Reg/Strategy"              value="0"/>
    <param name="RGBD/CreateOccupancyGrid" value="true"/>
    <param name="Grid/3D" value="false"/>
    <param name="Grid/Sensor" value="0.1"/>
    <param name="Grid/MapFrame" value="map"/>
    <param name="Grid/CellSize"         value="0.05"/>
    <param name="Grid/RangeMin" value="0.05"/>
    <param name="Grid/RangeMax" value="5.0"/>

    <param name="RGBD/ProximityPathMaxNeighbors" type="string" value="10"/>
    <param name="Reg/IcpInliersRatio" type="string" value="0.3"/>
    <param name="Icp/MaxIterations" type="string" value="100"/> -->
    <!-- <param name="RGBD/ResetCountdown" value="1"/> -->
  </node>

  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
      <remap from="map" to="/grid_map" />
      <param name="base_local_planner" value="teb_local_planner/TebLocalPlannerROS" />

      <!-- <param name="base_global_planner" value="global_planner/GlobalPlanner" /> -->
      
      <rosparam file="$(find ros_x_habitat)/navigation_params/move_base_params.yaml" command="load" />

      <rosparam file="$(find ros_x_habitat)/navigation_params/global_costmap.yaml" command="load" ns="global_costmap" />
      <rosparam file="$(find ros_x_habitat)/navigation_params/local_costmap.yaml" command="load" ns="local_costmap" />
      <!--<rosparam file="$(find ros_x_habitat)/navigation_params/base_local_planner_params.yaml" command="load" />-->
      <rosparam file="$(find ros_x_habitat)/navigation_params/teb_local_planner_params.yaml" command="load" />

      <!-- <rosparam file="$(find ros_x_habitat)/navigation_params/global_planner_params.yaml" command="load" /> -->
  </node>

  <node pkg="ros_x_habitat" type="distance_tracker.py" name="distance_tracker" output="screen" />
</launch>
